lib/base.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const debug = require('debug')('hmpo:stubber:base');
+const path = require('path');
+
+class Base {
+    constructor(options, name, basePath) {
+        this.name = name;
+        this.options = options || {};
+        this.basePath = basePath;
+        this.lastBasePath = basePath;
+    }
+
+    // Pick values from req params based on the locations specified in options
+    param(req, options, def) {
+        let item =
+            (options.query && req.query && req.query[options.query]) ||
+            (options.param && req.params && req.params[options.param]) ||
+            (options.body && req.body && req.body[options.body]) ||
+            (options.header && req.get(options.header)) ||
+            (options.session && req.session && req.session[options.session]) ||
+            (options.env && process.env[options.env]) ||
+            def;
+        return item;
+    }
+
+    // If value is a string, require() it relative to the current basePath
+    // Set the lastBasePath to the dirname of the resolved file
+    require(value, defaultValue) {
+        if (typeof value === 'string') {
+            let filename = value;
+            value = null;
+            debug('Resolving filename', this.name, filename);
+            this.lastBasePath = this.basePath;
+            filename = require.resolve(path.resolve(this.basePath, filename));
+            debug('Requiring filename', this.name, filename);
+            value = require(filename);
```

Comment:
Yes I think this needs to change to catch and re-throw an error. require.resolve with throw if the file is not found. require will then throw if there is a syntax error.